### PR TITLE
upsample_*.vec ops are now CompositeImplicit

### PR DESCRIPTION
**Pull request by fdrocha**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #85638

It was previously CompositeExplicit but it was not really necessary.
See discussion in https://github.com/pytorch/pytorch/issues/85405

cc @jerryzh168 @jianyuh @raghuramank100 @jamesr66a @vkuzo @jgong5 @Xia-Weiwen @leslie-fang-intel @VitalyFedyunin @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @Guobing-Chen @chunyuan-w @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

**Review by ezyang — APPROVED** [review]
Okey dokey

**Review by lezcano — APPROVED** [review]
If CI is happy, I am happy.

**Comment by zou3519** [comment]
to fix vmap, does adding `OP_DECOMPOSE2(upsample_bicubic2d, vec)` into https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/functorch/BatchRulesDecompositions.cpp work?

**Comment by fdrocha** [comment]
> to fix vmap, does adding `OP_DECOMPOSE2(upsample_bicubic2d, vec)` into https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/functorch/BatchRulesDecompositions.cpp work?

Unfortunately no, because `aten::upsample_bicubic2d_vec` no longer exists (only the version without the `_vec`) so compilation fails.
I think I managed to fix it in latest commit, but needs to be cleaned up a bit.

**Comment by lezcano** [comment]
>Unfortunately no, because aten::upsample_bicubic2d_vec no longer exists (only the version without the _vec)

What do you mean? `bicubic2d_vec` exists, only that it's CompositeImplicit. What does not exist is its backward, but that's alright.

**Comment by fdrocha** [comment]
Sorry, what I said before was not really right.

@zou3519, I had misread your comment. If I add `OP_DECOMPOSE2(upsample_bicubic2d, vec)` (and change `UPSAMPLE_BATCH(upsample_bicubic2d)` to `EXISTING_BDIM(upsample_bicubic2d)` above to avoid a warning about duplicate registration), it does compile correctly, but then it fails tests like `test/functorch/test_ops.py::TestOperatorsCPU::test_vmapvjp_has_batch_rule_nn_functional_interpolate_bicubic_cpu_float32` with

```
E           RuntimeError: aten::upsample_bicubic2d_backward hit the vmap fallback which is currently disabled
```

@lezcano,  I mean't `upsample_bicubic2d_backward_vec` doesn't exist. If I don't change this file and keep ` UPSAMPLE_BACKWARD(upsample_bicubic2d_backward, vec)` in there, compilation fails because

```
: error: 'upsample_bicubic2d_backward_vec_generated_plumbing' was not declared in this scope; did you mean 'upsample_bicubic2d_backward_generated_plumbing'
```

The latest commit fixes these problems, but maybe there is a nicer way of doing it.
Or alternatively, maybe I can make the same change for all the `vec` overloads (i.e., make them composite implicit) and then it is nicer and more uniform.

**Comment by fdrocha** [comment]
I have fixed all the other `upsample*.vec` operators, which had the same issue. One even had separate cuda and cpu implementations that did exactly the same thing.

There are a few other variants, that start with `_upsample`, that could also be cleaned up. If nothing new turns up in CI for this commit, will try that next.

File: aten/src/ATen/native/native_functions.yaml
Comment (lezcano) [inline aten/src/ATen/native/native_functions.yaml:10971]: There must be also a `upsample_nearest3d_quantized_cpu` to remove, right?

File: aten/src/ATen/native/native_functions.yaml
Comment (lezcano) [inline aten/src/ATen/native/native_functions.yaml:10922]: Also, what is it with these?

File: aten/src/ATen/native/native_functions.yaml
Comment (fdrocha) [reply aten/src/ATen/native/native_functions.yaml:10971]: Right you are, will remove that.

File: aten/src/ATen/native/native_functions.yaml
Comment (fdrocha) [reply aten/src/ATen/native/native_functions.yaml:10922]: `interpolate` will call different versions of all of these guys depending on what params you pass it. E.g,

https://github.com/pytorch/pytorch/blob/30fb2c4abaaaa966999eab11674f25b18460e609/torch/csrc/api/include/torch/nn/functional/upsampling.h#L145

I will clean up these next, once I can check that there's no unexpected problem from the changes so far.

**Comment by fdrocha** [comment]
@wonjoolee95 could you please take a look? The failing check is the XLA tests which fails with

```
AssertionError: Found an invalid operator name: upsample_nearest2d_backward.vec
``` 

This PR makes `upsample_nearest2d.vec` CompositeImplicit, so the backward does not have an explicit implementation anymore. Hopefully it's an easy fix on the XLA side. 
There probably will be similar errors for other `upsample_*_backward.vec` operators once that one is fixed.

**Comment by wonjoolee95** [comment]
> @wonjoolee95 could you please take a look? The failing check is the XLA tests which fails with
> 
> ```
> AssertionError: Found an invalid operator name: upsample_nearest2d_backward.vec
> ```
> 
> This PR makes `upsample_nearest2d.vec` CompositeImplicit, so the backward does not have an explicit implementation anymore. Hopefully it's an easy fix on the XLA side. There probably will be similar errors for other `upsample_*_backward.vec` operators once that one is fixed.

Sure thing, thanks for letting me know, @fdrocha. So are we safe to remove the `upsample_*_backward.vec` ops on XLA's side with this PR?

**Comment by wonjoolee95** [comment]
> @wonjoolee95 could you please take a look? The failing check is the XLA tests which fails with
> 
> ```
> AssertionError: Found an invalid operator name: upsample_nearest2d_backward.vec
> ```
> 
> This PR makes `upsample_nearest2d.vec` CompositeImplicit, so the backward does not have an explicit implementation anymore. Hopefully it's an easy fix on the XLA side. There probably will be similar errors for other `upsample_*_backward.vec` operators once that one is fixed.

Thanks for letting me know. I'll have a fix on XLA's side and link it here soon.

**Comment by wonjoolee95** [comment]
@fdrocha, could we rebase the base branch for this PR with latest PyTorch master?

**Comment by fdrocha** [comment]
> @fdrocha, could we rebase the base branch for this PR with latest PyTorch master?

Done.

**Comment by fdrocha** [comment]
Actually, this triggered a bunch of test failures. Working through them, will let you know when done.

**Comment by fdrocha** [comment]
@wonjoolee95 it should be good now

**Comment by wonjoolee95** [comment]
Thanks, @fdrocha. The CI on XLA PR is still running but build/tests are passing locally. Could you update the https://github.com/pytorch/pytorch/blob/gh/fdrocha/15/base/.github/ci_commit_pins/xla.txt to `3bd2d21ee6874010c95de2f2d2e6660afc47f369` on this PR?

**Comment by fdrocha** [comment]
Thanks @wonjoolee95, I updated the pin.

**Review by malfet — APPROVED** [review]

**Comment by wonjoolee95** [comment]
@fdrocha, can we actually wait a little bit on merging this? We're verifying some regression changes on XLA's side (although CI's are green). I should be able to wrap it up on XLA's side by later today.

**Comment by fdrocha** [comment]
> @fdrocha, can we actually wait a little bit on merging this? We're verifying some regression changes on XLA's side (although CI's are green). I should be able to wrap it up on XLA's side by later today.

Of course, I canceled the merge. Will wait to hear from you before I try again.

**Comment by wonjoolee95** [comment]
On XLA's side, we're still seeing some of our existing unit tests fail. We're actively looking into it.

**Comment by lezcano** [comment]
@fdrocha now that Torchdynamo has been merged into master you may need to add the relevant changes from your other PR to this one.

**Review by jansel — APPROVED** [review]

**Comment by fdrocha** [comment]
@wonjoolee95 any luck with the XLA side? Would like to merge this if possible.

cc @JackCaoG

**Comment by JackCaoG** [comment]
Sorry for the late reply, we are still somewhat stuck at the pytorch/xla pr. There is a weird error, let's move the discussion to the xla pr in https://github.com/pytorch/xla/pull/4087

**Comment by wonjoolee95** [comment]
@fdrocha, I think the XLA's PR should be ready. Can you update rebase this PR and update the XLA commit pin to `1d0c3393fb48cb8740379e4fea9c37a0e131a7dd`?

**Comment by fdrocha** [comment]
> @fdrocha, sorry for the delay, I think the XLA's PR should be ready now. Can you rebase this PR and update the XLA commit pin to `1d0c3393fb48cb8740379e4fea9c37a0e131a7dd`?

Thank you @wonjoolee95. Just pushed the update, fingers crossed this it!

**Comment by fdrocha** [comment]
Guess it wasn't it
I will sort through these failures as soon as I can, not sure if I'll be able to get to it today.

**Comment by wonjoolee95** [comment]
Thank you, @fdrocha. XLA PR's CI is looking good. We should be good to merge this when this CI passes. Please @ me once  before you trigger the merge for just a heads-up. Thanks!

**Comment by fdrocha** [comment]
@wonjoolee95 this took me longer than I expected, but is good to go now (the failing test comes from upstream).
Please confirm that it is okay to go ahead and merge.

**Comment by JackCaoG** [comment]
I think we need to rebase xla pr(ideally this pr too otherwise xla ci in this pr will fail but should pass on master) before merging, there was an offending pr merged in yesterday.

**Comment by wonjoolee95** [comment]
Thanks. Just rebased XLA's PR on master, the latest hash pin is `7889d2d3be16675943d84e4a4133ed7c245a623f`. 

@fdrocha, can we rebase this PR one final time with the updated pin and trigger merge?

**Comment by fdrocha** [comment]
@pytorchmergebot -f "Test failures are CI issues"

cc @wonjoolee95

**Comment by fdrocha** [comment]
@pytorchbot  merge -f "Test failures are CI issues"

cc @wonjoolee95

**Pull request closed without merging**
